### PR TITLE
feat: switch to ui-avatars for default avatar

### LINKED ISSUE #10475: [CAL-2273] Use ui-avatar.com only for all default avatars
Possible Followup: Avatar component has fallback image support but it uses [gravatar.com](http://gravatar.com) maybe we can just switch it with [ui-avatars.com](http://ui-avatars.com) and automatically enable fallback.

There is one concern that sometimes we show a initials based avatar and sometimes the silhouette avatar.

\*Originally posted by @hariombalhara in \*[*https://github.com/calcom/cal.com/pull/10383#discussion_r1275991923*](https://github.com/calcom/cal.com/pull/10383#discussion_r1275991923)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2273](https://linear.app/calcom/issue/CAL-2273/use-ui-avatarcom-only-for-all-default-avatars)</sub>

### PR DESCRIPTION
**Pull request by Saga-sanga**
## What does this PR do?

- Updated `defaultAvatarSrc` method in `defaultAvatarImage.ts` to add a default image from [ui-avatars.com](http://ui-avatars.com/)  instead of [gravatar.com](http://gravatar.com/).
- Set up the method to use the `alt` text passed to `Avatar` component to generate the image.
- Preserved the use of silhouettes where previously used and used initials where [gravatar.com](http://gravatar.com/) was used. 
 

/claim #10475

Fixes #10475

**Before:**
Screenshot 2023-08-07 at 3 00 10 PM

**After:**
Screenshot 2023-08-07 at 2 58 21 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by Saga-sanga — COMMENTED** [review]
self reviewed

**Review by Udit-takkar — COMMENTED** [review]
@Saga-sanga do not push yarn.lock if you have not made any changes in package.json.

Edit: I have removed it now

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
Screenshot 2023-08-04 at 6 34 50 PM

i see black images in some pages

**Comment by Saga-sanga** [comment]
> Screenshot 2023-08-04 at 6 34 50 PM
> i see black images in some pages

Yes, that's probably because the images have no alt value. That's because [ui-avatars.com](ui-avatars.com) expects a string value to generate the images. A possible fix would be to add a default string value if there is no alt value.

**Comment by Saga-sanga** [comment]
@Udit-takkar could you please re-review my PR?

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
1) You have not updated avatar.stories.mdx

2)  Here on onboarding step. For a user we should display default avatar icon because that's what i see in booking page and other pages.

Screenshot 2023-08-10 at 2 11 29 PM

this default icon should be displayed

Screenshot 2023-08-10 at 2 12 41 PM

3) api/user/avatar route still uses defaultAvatarSrc which as www.gravatar.com

i think instead of removing gravatarFallbackMd5 @hariombalhara meant to pass [ui-avatars.com](http://ui-avatars.com/) to gravatar.com in defaultAvatarSrc . can you confirm once ? @hariombalhara  

Screenshot 2023-08-10 at 2 28 32 PM

**Comment by Saga-sanga** [comment]
@Udit-takkar Hi, what exactly do I need to update in `avatar.stories.mdx`? I am not familiar with storybook so some feedback would be helpful.

File: packages/app-store/apps.metadata.generated.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ import ga4_config_json from "./ga4/config.json";
 import { metadata as giphy__metadata_ts } from "./giphy/_metadata";
 import { metadata as googlecalendar__metadata_ts } from "./googlecalendar/_metadata";
 import { metadata as googlevideo__metadata_ts } from "./googlevideo/_metadata";
+import greet_meet_config_json from "./greet-meet/config.json";
```
Comment (hariombalhara) [inline packages/app-store/apps.metadata.generated.ts:24]: This looks like an unrelated change

Comment (Saga-sanga) [reply packages/app-store/apps.metadata.generated.ts:24]: Yes, I'll remove it. I was playing around with the app creator and the changes persisted across branches.

**Comment by hariombalhara** [comment]
I discussed internally and we agreed that everywhere for a user we should use gravatar(silhoutte) and for non-user(team/org) we use initials based avatar.

I was thinking that maybe we can specify the type of fallback to Avatar component as a prop(e.g. fallbackTo=gravatar|initials) and then Avatar component can take care of showing the appropriate fallback instead of we passing the `gravatarFallbackMd5` or calling `getPlaceholderAvatar` for every usage of Avatar.

Also, we can then create HOCs like <TeamAvatar/> <UserAvatar/> <OrgAvatar/> that can be used throughout the app

What do you think @Saga-sanga @sean-brydon ?
